specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2018-06-17-preview/workbooks_API.json

```diff
@@ -0,0 +1,568 @@
+    "WorkbooksListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Workbook"
+          },
+          "description": "An array of workbooks."
+        }
+      },
+      "description": "Workbook list result."
+    },
+    "Workbook": {
+      "description": "An Application Insights workbook definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a workbook for an Azure resource.",
+          "$ref": "#/definitions/WorkbookProperties"
+        }
+      }
+    },
+    "WorkbookProperties": {
+      "description": "Properties that contain a workbook.",
+      "required": [
+        "displayName",
+        "category",
+        "serializedData",
+        "userId"
+      ],
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The user-defined name (display name) of the workbook."
+        },
+        "serializedData": {
+          "type": "string",
+          "description": "Configuration of this particular workbook. Configuration data is a string containing valid JSON"
+        },
+        "timeModified": {
+          "type": "string",
+          "description": "Date and time in UTC of the last modification that was made to this workbook definition.",
+          "readOnly": true
+        },       
+        "category": {
+          "type": "string",
+          "description": "Workbook category, as defined by the user at creation time."
+        },
+        "version": {
+          "type": "string",
+          "description": "Workbook version"
+        },
+        "tags": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of 0 or more tags that are associated with this workbook definition"
+        },
+        "userId": {
+          "type": "string",
+          "description": "Unique user id of the specific user that owns this workbook."
+        }
+      }
+    },
+    "WorkbookUpdateParameters": {
+      "properties": {
+        "kind": {
+            "type": "string",
+            "description": "The kind of workbook. Choices are user and shared.",
+            "enum": [
+                "user",
+                "shared"
+            ],
+            "x-ms-enum": {
+                "name": "SharedTypeKind",
+                "modelAsString": true
+            }
+        },
+        "tags": {
+            "type": "object",
+            "additionalProperties": {
+                "type": "string"
+            },
+            "description": "Resource tags"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
```

Comment:
Can user change the location of workbook resource? if not please remove this property.

Reply:
@anuchandy  Done!